### DISCUSSION TITLE
ERROR: No matching distribution found for gruut[cs,de,es,fr,it,nl,pt,ru,sv]~=1.2.0

### DISCUSSION THREAD

**Discussion by tiomaldy** [coqui-ai/TTS#632] [General Q&A]
Help for solve this i can t find anything about this

**Comment by tiomaldy** [comment]
ERROR: Command errored out with exit status 1:
     command: 'f:\python38\enviroment\scripts\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Gustavo\\AppData\\Local\\Temp\\pip-install-95kitrcw\\gruut_985c33e6e2af4321a5b9d1f920ed138f\\setup.py'"'"'; __file__='"'"'C:\\Users\\Gustavo\\AppData\\Local\\Temp\\pip-install-95
kitrcw\\gruut_985c33e6e2af4321a5b9d1f920ed138f\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec
'"'"'))' egg_info --egg-base 'C:\Users\Gustavo\AppData\Local\Temp\pip-pip-egg-info-b9_4jnmw'
         cwd: C:\Users\Gustavo\AppData\Local\Temp\pip-install-95kitrcw\gruut_985c33e6e2af4321a5b9d1f920ed138f\
    Complete output (9 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\Gustavo\AppData\Local\Temp\pip-install-95kitrcw\gruut_985c33e6e2af4321a5b9d1f920ed138f\setup.py", line 15, in <module>
        long_description = readme_path.read_text()
      File "F:\Python38\lib\pathlib.py", line 1215, in read_text
        return f.read()
      File "F:\Python38\lib\encodings\cp1252.py", line 23, in decode
        return codecs.charmap_decode(input,self.errors,decoding_table)[0]
    UnicodeDecodeError: 'charmap' codec can't decode byte 0x90 in position 2948: character maps to <undefined>
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/9d/3f/28b4146581fffc33d58abd973a95d421fda274871141f557d106e710bb74/gruut-0.2.1.tar.gz#sha256=37f06c2d8228cf33b4664a0d957b4066bdd8c3e7025ec28650d10b56de3860fd (from https://pypi.org/simple/gruut/) (requires-python:>=3.7). Command errored out with exit s
tatus 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement gruut (from versions: 0.2.1, 0.3.0, 0.7.0, 0.8.0, 0.9.1, 0.9.2, 0.9.3, 0.9.4, 0.9.5, 1.0.0, 1.1.0, 1.2.1, 1.2.2)
ERROR: No matching distribution found for gruut

**Reply by godspirit00** [reply]
This has something to do with the encoding.
I encountered this error too, and solved it by adding a line to the `pathlib.py` mentioned in the error message.

Add 
`        if encoding==None:
            encoding='utf-8'
`
before the `with` statement before the `return f.read()`  mentioned in the error message.

**Comment by erogol** [comment]
@synesthesiam 👀

**Comment by synesthesiam** [comment]
I got caught by [this common mistake](https://www.python.org/dev/peps/pep-0597/#using-the-default-encoding-is-a-common-mistake). The default encoding in Windows is apparently not UTF-8, so a README in a package with anything outside of ASCII will break on Windows without an explicit encoding specified.

I've pushed a new version of gruut (1.2.3) with this fixed. The new version should be picked up automatically on a clean install, or you can use `--upgrade` :slightly_smiling_face:

**Reply by tiomaldy** [reply]
Yes i solved yesterday thanks.

**Reply by RobinE89** [reply]
@synesthesiam 
> 
> 
> I got caught by [this common mistake](https://www.python.org/dev/peps/pep-0597/#using-the-default-encoding-is-a-common-mistake). The default encoding in Windows is apparently not UTF-8, so a README in a package with anything outside of ASCII will break on Windows without an explicit encoding specified.
> 
> I've pushed a new version of gruut (1.2.3) with this fixed. The new version should be picked up automatically on a clean install, or you can use `--upgrade` 🙂

same issue -> sound like this is what im looking for but if i use the command install gruut (or TTS) - then i keep getting the same error (now). I don't understand why pip install always changes to a different version, even if I install it from a local package:

(venv) .........>pip install ".....\Documents\**gruut-1.2.3.tar.gz**"
Processing .....\documents\gruut-1.2.3.tar.gz
Collecting Babel~=2.8.0
  Using cached Babel-2.8.1-py2.py3-none-any.whl (8.6 MB)
**Collecting gruut-ipa~=0.9.0   <- ???????????????????????
  Using cached gruut-ipa-0.9.2.tar.gz (36 kB)
    ERROR: Command errored out with exit status 1:** 
.....
**UnicodeDecodeError: 'charmap' codec can't decode byte 0x90 in position 337: character maps to <undefined>**

also on: pip install --no-cache-dir -I gruut==1.2.3
Collecting gruut==1.2.3......

**UnicodeDecodeError: 'charmap' codec can't decode byte 0x90 in position 337: character maps to <undefined>**

Hopefully you have some advice for me here ^^

**Reply by synesthesiam** [reply]
Looks like there was the same problem with gruut-ipa (a core dependency of gruut) -- I had the audacity to use assume UTF-8 in a README was OK in 2021 :stuck_out_tongue: 

I just pushed gruut-ipa 0.9.3, so I'm hoping that will fix your problem. Sorry for the confusion!

**Reply by RobinE89** [reply]
@synesthesiam 
thx a lot. its running now =)
